src/components/views/rooms/RoomPreviewBar.tsx

```diff
@@ -310,18 +315,12 @@ export default class RoomPreviewBar extends React.Component<IProps, IState> {
             }
             case MessageCase.NotLoggedIn: {
                 const opts: RoomPreviewOpts = { canJoin: false };
-                if (this.props.room?.roomId) {
-                    ModuleRunner.instance.invoke(
-                        RoomViewLifecycle.PreviewRoomNotLoggedIn,
-                        opts,
-                        this.props.room.roomId,
-                    );
-                }
+                ModuleRunner.instance.invoke(RoomViewLifecycle.PreviewRoomNotLoggedIn, opts, this.props.roomId);
                 if (opts.canJoin) {
                     title = _t("Join the room to participate");
                     primaryActionLabel = _t("Join");
                     primaryActionHandler = () => {
-                        ModuleRunner.instance.invoke(RoomViewLifecycle.JoinFromRoomPreview, this.props.room.roomId);
+                        ModuleRunner.instance.invoke(RoomViewLifecycle.JoinFromRoomPreview, this.props.roomId);
                     };
```

Comment:
question: how did this behave before your change? It seems like it would throw a type error if used without a `room` property.

Reply:
before it was an if statement around with `this.props.room?.roomId` and if true then `room.roomId` was passed to invoke, but invoke has params as `...args: any[]` so takes anything, it could fail in PR version if module implementation does not expect undefined for `roomId` but this was fixed with your request to change below, so seems fine now.